data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/service/ContentCacheUpdaterService.java

```diff
@@ -75,4 +83,24 @@ public void process(ContentCacheUpdaterSource source, ContentCacheUpdaterSink si
         }
         return contentData.getParsedData(parsedData, newProperties);
     }
-}
+
+    public void updateDialCodeToCache(List<String> dialCodeList, int storeId, String apiUrl) {
+        if (null != dialCodeList) {
+            dialCodeList.forEach(dialCode -> {
+                if (!dialCode.isEmpty()) {
+                    String metaData = readFromCache(dialCode, storeId);
+                    if (null == metaData || metaData.isEmpty()) {
+                        LOGGER.info("", String.format("Invoking DialCode API to fetch the metadata for this dialCode( %s )", dialCode));
+                        Object dialCodeMetadata = contentData.getMetadata(apiUrl.concat(dialCode), this.restUtil, contentCacheConfig.getAuthorizationKey(), "dialcode");
```

Comment:
What happens if the dialcode API is not responding? Do we do a retry? if so - how many times or do we stop the job?

Reply:
**What happens if the dialcode API is not responding?**  Do we do a retry?  - Unirest is internally handling retry logic.
**do we stop the job?** - Yes, Stopping the job by throwing an exception.